### ISSUE TITLE
overloading a function using declare is producing ESLint no-redeclare error

### ISSUE THREAD

**Issue by gajus** [gajus/eslint-plugin-flowtype#374]
```js
declare function DataPoolConnectMethodType (): Promise<DatabasePoolConnectionType>;
declare function DataPoolConnectMethodType ((connection: DatabaseIsolatedPoolConnectionType) => Promise<void>): Promise<void>;

```

Expected:

No error.

Actual:

no-redeclare error.

Related issues:

* https://eslint.org/docs/rules/no-redeclare
* https://github.com/babel/eslint-plugin-babel/issues/162

**Comment by bsonntag** [comment]
Any updates on this?

**Comment by gajus** [comment]
A PR is welcome, but no one is working on this.
